js/script.js

```diff
@@ -0,0 +1,328 @@
+gamehasended = false;
+firstCharacterTyped = false;
+
+function loadParagraph() {
+  const ranIndex = Math.floor(Math.random() * paragraphs.length);
+  typingText.innerHTML = "";
+  paragraphs[ranIndex].split("").forEach((char) => {
+    let span = `<span>${char}</span>`;
+    typingText.innerHTML += span;
+  });
+  typingText.querySelectorAll("span")[0].classList.add("active");
+  document.addEventListener("keydown", () => inpField.focus());
+  typingText.addEventListener("click", () => inpField.focus());
+}
+
+deleteToday.addEventListener("click", (e) => {
+  clearTodaysBest();
+});
+
+deleteAlltime.addEventListener("click", (e) => {
+  clearAlltimeBest();
+});
+
+playAgainButton.addEventListener("click", () => {
+  location.reload();
+});
+
+function toggleGame() {
+  if (gamehasended) {
+    print;
+    gameFrame.style.display = "none";
+    resultFrame.style.display = "block";
+  } else {
+    resultFrame.style.display = "none";
+  }
+  document.addEventListener("keydown", () => inpField.focus());
+  typingText.addEventListener("click", () => inpField.focus());
+}
+
+async function generateWords() {
+  fetch(
+    "https://random-word-api.herokuapp.com/word?number=" +
+      document.getElementById("word-count").value
+  )
+    .then((res) => res.json())
+    .then((data) => (output = data.toString().replaceAll(",", " ")))
+    .then((output) => (typingText.innerHTML = output))
+    .then(() => splitWords());
+  typingText.focus();
+
+  clearInterval(intervalId);
+  timeSpend = 0;
+  firstCharacterTyped = false;
+
+  wordCounter.style.display = "none"
+  generateWordButton.style.display = "none"
+}
+
+let myArray;
+
+function splitWords() {
+  let text = typingText.innerHTML;
+  typingText.innerHTML = "";
+  myArray = text.split("");
+  myArray.forEach((span) => {
+    let spanTag = `<span class="test">${span}</span>`;
+    typingText.innerHTML += spanTag;
+  });
+}
+
+function Typing() {
+  const characters = typingText.innerText;
+  const typedCharactersNumber = document.getElementsByClassName("test");
+  const numberOfWritenCharacters =
+    document.getElementsByClassName("typed").length;
+  let typedChar = inpField.value.split("")[charIndex];
+  const character = characters[charIndex];
+  const list = typedCharactersNumber[charIndex].classList;
+  const numberOfCorrectCharacters =
+    document.getElementsByClassName("correct").length;
+
+  if (firstCharacterTyped == false) {
+    firstCharacterTyped = true;
+    intervalId = window.setInterval(function () {
+      timeSpend++;
+    }, 1000);
+  }
+
+  if (character === typedChar) {
+    if (typedChar == " ") {
+      list.remove("active");
+      list.add("correct");
+      list.add("typed");
+    }
+    list.remove("active");
+    list.add("correct");
+    list.add("typed");
+  } else {
+    list.remove("active");
+    list.add("incorrect");
+    list.add("typed");
+    mistakes++;
+  }
+
+  charIndex++;
+  if (numberOfWritenCharacters + 1 == characters.length) {
```

Comment:
```suggestion
  if (numberOfWrittenCharacters + 1 == characters.length) {
```
Also a magic number here "1"